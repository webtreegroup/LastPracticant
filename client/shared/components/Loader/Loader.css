.loader {
    background: #00000075;
    position: fixed;
    left: 0;
    right: 0;
    bottom: 0;
    top: 0;
    transition: 300ms;
    z-index: 9000;
    visibility: hidden;
    opacity: 0;

    &_visible {
        visibility: visible;
        opacity: 1;
    }

    &__img {
        width: 55px;
        height: 32px;
        background: url(./loader.gif) no-repeat center/contain;
    }
}
