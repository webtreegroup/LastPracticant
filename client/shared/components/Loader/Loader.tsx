import React, { FC, memo } from 'react';
import './Loader.css';

import { ComponentCommonProps } from 'client/shared/types';
import bem from 'bem-cn';
import { NivelatorXY } from '../NivelatorXY';

interface LoaderProps extends ComponentCommonProps {
    isVisible: boolean
}

const block = bem('loader');

export const Loader: FC<LoaderProps> = memo(
    ({
        className,
        isVisible,
    }) => (
        <div className={block({ visible: isVisible }).mix(className)}>
            <NivelatorXY>
                <div className={block('img')} />
            </NivelatorXY>
        </div>
    ),
);
