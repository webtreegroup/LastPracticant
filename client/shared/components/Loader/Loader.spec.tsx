import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Loader } from './Loader';

afterEach(cleanup);

describe('Loader', () => {
    it('Render hide Loader component', () => {
        const { asFragment } = render(<Loader isVisible={false} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Render show Loader component', () => {
        const { asFragment } = render(<Loader isVisible />);
        expect(asFragment()).toMatchSnapshot();
    });
});
