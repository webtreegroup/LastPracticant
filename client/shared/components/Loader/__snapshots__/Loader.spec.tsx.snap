// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Render hide Loader component 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    <div
      class="MuiGrid-root nivelator-xy MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <img
        alt="loader"
        src="[object Object]"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Loader Render show Loader component 1`] = `
<DocumentFragment>
  <div
    class="loader loader_visible"
  >
    <div
      class="MuiGrid-root nivelator-xy MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <img
        alt="loader"
        src="[object Object]"
      />
    </div>
  </div>
</DocumentFragment>
`;
